packages/react-player/src/react-player.tsx

```diff
@@ -210,159 +150,106 @@ export const DotLottiePlayer: React.FC<DotLottiePlayerProps> = ({
       case PlayerState.Error:
-        onError?.();
+        onEvent?.(PlayerEvents.Error);
         break;
 
       default:
         break;
     }
   }, [currentState]);
 
-  function renderControls(): JSX.Element {
-    return (
-      <div aria-label="lottie-animation-controls" className="toolbar">
-        <button
-          onClick={togglePlay}
-          className={`${isPlaying || isPaused ? 'active' : ''}`}
-          style={{ alignItems: 'center' }}
-          // tabindex={0}
-          aria-label="play-pause"
-        >
-          {isPlaying ? (
-            <svg width="24" height="24" aria-hidden="true" focusable="false">
-              <path d="M14.016 5.016H18v13.969h-3.984V5.016zM6 18.984V5.015h3.984v13.969H6z" />
-            </svg>
-          ) : (
-            <svg width="24" height="24" aria-hidden="true" focusable="false">
-              <path d="M8.016 5.016L18.985 12 8.016 18.984V5.015z" />
-            </svg>
-          )}
-        </button>
-        <button
-          onClick={(): void => dotLottiePlayer?.stop()}
-          className={`${isStopped ? 'active' : ''}`}
-          style={{ alignItems: 'center' }}
-          aria-label="stop"
-        >
-          <svg width="24" height="24" aria-hidden="true" focusable="false">
-            <path d="M6 6h12v12H6V6z" />
-          </svg>
-        </button>
-        <input
-          className="seeker"
-          type="range"
-          min={0}
-          step={0}
-          max={100}
-          value={seeker || 0}
-          onInput={(event): void => handleSeekChange(event)}
-          onMouseDown={(): void => {
-            setPrevState(currentState);
-            dotLottiePlayer?.freeze();
-          }}
-          onMouseUp={(): void => {
-            seek(frame || 0, _prevState);
-          }}
-          aria-valuemin={1}
-          aria-valuemax={100}
-          role="slider"
-          aria-valuenow={seeker}
-          aria-label="lottie-seek-input"
-        />
-        <button
-          onClick={(): void => {
-            setLooping(!isLoop);
-          }}
-          className={isLoop ? 'active' : ''}
-          style={{ alignItems: 'center' }}
-          aria-label="loop-toggle"
-        >
-          <svg width="24" height="24" aria-hidden="true" focusable="false">
-            <path d="M17.016 17.016v-4.031h1.969v6h-12v3l-3.984-3.984 3.984-3.984v3h10.031zM6.984 6.984v4.031H5.015v-6h12v-3l3.984 3.984-3.984 3.984v-3H6.984z" />
-          </svg>
-        </button>
-      </div>
-    );
-  }
-
   return (
-    <div
-      className={`dotlottie-container main ${controls ? 'controls' : ''} ${className}`}
-      lang="en"
-      role="img"
-      {...(testId && {
-        'data-testid': testId,
-        'data-player-state': currentState,
-        'data-player-seeker': seeker || 0,
-      })}
-      {...props}
+    <DotLottieContext.Provider
+      value={{
+        currentState,
+        dotLottiePlayer,
+        frame,
+        seeker,
+        loop: Boolean(loop),
+      }}
```

Comment:
I believe providing the Context value as an anonymous object may not be the most efficient approach. If a component re-renders, even when there have been no changes in the Context value, it would result in all the Context consumers re-rendering as well. This is due to the fact that a new reference is generated for the value object each time the component is rendered.

I believe we need to memoize the context value.